### PR TITLE
[Communication]: Identifier serializer update for new model

### PR DESCRIPTION
**Pull request by JianpingChen**

### CODE REVIEW

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java [hunk 1]
```diff
@@ -3,77 +3,106 @@
...
+    /**
+     * Deserialize CommunicationIdentifierModel into CommunicationIdentifier
+     * @param identifier CommunicationIdentifierModel to be deserialized
+     * @return deserialized CommunicationIdentifier
+     */
     public static CommunicationIdentifier deserialize(CommunicationIdentifierModel identifier) {
-        String id = identifier.getId();
-        CommunicationIdentifierKind kind = identifier.getKind();
+        assertSingleType(identifier);
+        String rawId = identifier.getRawId();
```
Comment (RezaJooyandeh) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java:19]: Do not we need a `requireNonNull` here as well?

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
     /**
      * Creates a CommunicationUserIdentifier object
      *
-     * @param id the string identifier representing the identity
+     * @param id id of the communication user.
```
Comment (RezaJooyandeh) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java:18]: >id id of

➡ "id of"

**Review by RezaJooyandeh — APPROVED** [review]

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java (hunk 2 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java:18]: The first "id" is the parameter name. What about "@param id identifier string of the Communication user"

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java (hunk 1 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java:19]: To my understanding, rawId is only required for TeamsUser and UnknownUser. For phone number and communication user, it is optional. Am I missing anything?

Comment (RezaJooyandeh) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java:20]: In the current schema `rawId` is required only for those two, but as a contract we want the services to always return it. So let's make it mandatory in the deserialization

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java (hunk 2 above)
Comment (RezaJooyandeh) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationUserIdentifier.java]: oh I had misunderstood how the comments work in Java. I think we are good.

**Pull request merged**